src/EditorFeatures/Core/Implementation/Diagnostics/AbstractDiagnosticsTaggerProvider.AggregatingTagger.cs

```diff
@@ -74,21 +84,21 @@ private class AggregatingTagger : ForegroundThreadAffinitizedObject, IAccurateTa
                 _owner._diagnosticService.DiagnosticsUpdated += OnDiagnosticsUpdated;
             }
 
-            private void GetInitialDiagnosticsInBackground(
-                Document document, CancellationToken cancellationToken)
+            private void GetInitialDiagnosticsInBackground(Document document)
             {
                 this.AssertIsBackground();
-                cancellationToken.ThrowIfCancellationRequested();
+                _shutdownCancellationSource.Token.ThrowIfCancellationRequested();
 
                 if (document != null)
                 {
                     var project = document.Project;
                     var workspace = project.Solution.Workspace;
-                    foreach (var updateArgs in _owner._diagnosticService.GetDiagnosticsUpdatedEventArgs(workspace, project.Id, document.Id, cancellationToken))
+                    foreach (var updateArgs in _owner._diagnosticService.GetDiagnosticsUpdatedEventArgs(workspace, project.Id, document.Id, _shutdownCancellationSource.Token))
                     {
-                        var diagnostics = AdjustInitialDiagnostics(project.Solution, updateArgs, cancellationToken);
-                        if (diagnostics.Length == 0)
+                        var diagnostics = AdjustInitialDiagnostics(project.Solution, updateArgs, _shutdownCancellationSource.Token);
+                        if (diagnostics.Length == 0 || !diagnostics.Any(d => _owner.IncludeDiagnostic(d)))
```

Comment:
what is this code for?

Reply:
so that unnecessary tagger doesn't enqueue task for diagnostics that are not related to unnecessary tagger.